File: TestTask/Core/Entities/Order/Order.cs
using Core.Entities.Enums;

namespace Core.Entities.Order;

public class Order : BaseEntity
{
    public Order()
    {
    }

    public Order(int customerId, IReadOnlyList<OrderItem> orderItems, OrderStatus status, string comment) 
    {
        CustomerId = customerId;
        Comment = comment;
        Status = status;
        OrderItems = orderItems;
    }
    public DateTime CreationDate { get; set; } = DateTime.UtcNow;
    public Customer Customer { get; set; }
    public int CustomerId { get; set; }
    public OrderStatus Status { get; set; } = OrderStatus.New;
    public string Comment { get; set; }
    public IReadOnlyList<OrderItem> OrderItems { get; set; } 
    public float GetTotal() => OrderItems.Sum(oi => oi.Price * oi.Quantity);
}

File: TestTask/frontend/src/app/models/order.ts
export interface OrderItem {
  productId: number;
  productName: string;
  size: string;
  category: string;
  price: number;
  quantity: number;
}

export interface Order {
  id: number;
  creationDate: string;
  customerName: string;
  customerAddress: string;
  status: string;
  comment: string;
  total: number;
  orderItems: OrderItem[];
}

export interface OrderToSend {
  customerId: number;
  status: string;
  comment: string;
  orderItems: OrderItem[];
}

export interface OrderItemToSend {
  productId: number;
  size: string;
  quantity: number;
}

export const TaskStatusValues = [
  { displayName: 'To Do', value: 0 },
  { displayName: 'In Progress', value: 1 },
  { displayName: 'Done', value: 2 },
  { displayName: 'Canceled', value: 3 },
];
